sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -172,20 +172,16 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper with Jo
       if (j.joinType == newJoinType) f else Filter(condition, j.copy(joinType = newJoinType))
 
     case a @ Aggregate(_, _, join @ Join(left, _, LeftOuter, _, _))
-        if a.isDistinct && a.references.subsetOf(AttributeSet(left.output)) &&
-          !canPlanAsBroadcastHashJoin(join, conf) =>
+        if a.isDistinct && a.references.subsetOf(AttributeSet(left.output)) =>
```

Comment:
nit: since we support `select a, a ...`, how about we rename `isDistinct` to `groupOnly`?

Reply:
OK.